Source/Particles/ParticleBoundaryBuffer.cpp

```diff
@@ -158,7 +200,19 @@ void ParticleBoundaryBuffer::gatherParticles (MultiParticleContainer& mypc,
                         amrex::Real phi_value  = DistanceToEB::interp_distance(ii, jj, kk, W, phiarr);
                         return phi_value < 0.0 ? 1 : 0;
                     },
-                                                    0, dst_index, np);
+                            [=] AMREX_GPU_HOST_DEVICE (const DstData& dst, const SrcData& src,
+                                                       int src_i, int dst_i) noexcept
+                        {
+                            dst.m_aos[dst_i] = src.m_aos[src_i];
+                            for (int j = 0; j < SrcData::NAR; ++j)
+                                dst.m_rdata[j][dst_i] = src.m_rdata[j][src_i];
+                            for (int j = 0; j < src.m_num_runtime_real; ++j)
+                                dst.m_runtime_rdata[j][dst_i] = src.m_runtime_rdata[j][src_i];
+                            for (int j = 0; j < src.m_num_runtime_int; ++j)
+                                dst.m_runtime_idata[j][dst_i] = src.m_runtime_idata[j][src_i];
+                            dst.m_runtime_rdata[timestamp_index][dst_i] = time;
+                        },
```

Comment:
This lambda looks to be identical in both call sites to `amrex::filterAndTransformParticles`. As such, could I suggest making a named functor our of it, and passing an instance of that? It looks like the only variable it actually captures is `time`

Reply:
Good idea, done.